tools/rex-extrude.c

```diff
@@ -0,0 +1,464 @@
+        float y = points[3 * i + 1];
+        float z = points[3 * i + 2];
+
+        //set vertex coordinates
+        mesh->positions[3 * numpoints + 3 * i]     = x + dx;
+        mesh->positions[3 * numpoints + 3 * i + 1] = y + dy;
+        mesh->positions[3 * numpoints + 3 * i + 2] = z + dz;
+
+        //set normals coordinates (optional)
+
+        //set textures coordinates (optional)
+
+        //set colors block (optional)
+
+        //each point creates four triangles, two for each orientation
+
+        if (closed == 0 && i == numpoints - 1) //if not closed, skip last point's triangles
+            break;
+
+        //upside down interior next triangle
+        mesh->triangles[triangle_vertex_id] = i; //current point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = i + numpoints; //current extruded point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1 + numpoints)%numpoints + numpoints; //next extruded point
+        triangle_vertex_id++;
+
+        //downside up interior next triangle
+        mesh->triangles[triangle_vertex_id] = i; //current point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1 + numpoints)%numpoints + numpoints; //next extruded point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1)%numpoints; //next point
+        triangle_vertex_id++;
+
+        //upside down exterior next triangle
+        mesh->triangles[triangle_vertex_id] = i; //current point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1 + numpoints)%numpoints + numpoints; //next extruded point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = i + numpoints; //current extruded point
+        triangle_vertex_id++;
+
+        //downside up exterior next triangle
+        mesh->triangles[triangle_vertex_id] = i; //current point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1)%numpoints; //next point
+        triangle_vertex_id++;
+        mesh->triangles[triangle_vertex_id] = (i+1 + numpoints)%numpoints + numpoints; //next extruded point
+        triangle_vertex_id++;
+    }
+
+    return mesh;
+}
+
+/**
+ * @brief Creates a Rex pointlist given an array of 3d points
+ */
+struct rex_pointlist* create_anchors (float *anchorpoints, uint32_t numanchors)
+{    
+    struct rex_pointlist* pointlist = malloc (sizeof (struct rex_pointlist));
+    if (!pointlist) //TODO should we die or just critical?
+        die ("Allocating pointlist failed");
+    rex_pointlist_init (pointlist);
+
+    pointlist->nr_vertices = numanchors;
+    pointlist->nr_colors = 0;
+
+    pointlist->positions = malloc (3 * numanchors * sizeof (float));
+    memcpy (pointlist->positions, anchorpoints, 3 * numanchors * sizeof (float));
+
+    return pointlist;
+}
+
+/**
+ * @brief Print the givne mesh in OFF format on stdout
+ */
+void off_print(struct rex_mesh *mesh)
```

Comment:
BTW, this would be a nice converter for REX files REX->OFF :-)

Reply:
Hehe, sure. Let's bookmark this and do it when lesser priority tasks are on the plate.

Reply:
Any way to label it as feature request somewhere?